pkg/state/state.go

```diff
@@ -1635,6 +1639,12 @@ func (st *HelmState) flagsForUpgrade(helm helmexec.Interface, release *ReleaseSp
 		flags = append(flags, "--cleanup-on-fail")
 	}
 
+	if helm.IsVersionAtLeast(3, 2) &&
```

Comment:
Nice! We do need `createNamespace` to be valid and configurable only when Helm is 3.2.0 or later. Thanks for doing this properly and elegantly.